lib/cloudant.js

```diff
@@ -580,6 +591,75 @@ Cloudant.prototype.ping = function(cb) {
 };
 
 /**
+ * Replace if the model instance exists with the same id or create a
+ * new instance
+ *
+ * @param {String} model The model name
+ * @param {Object} data The model instance data
+ * @param {Function} [cb] The callback function
+ */
+Cloudant.prototype.replaceOrCreate = function replaceOrCreate(
+  model, data, options, cb) {
+  debug('Cloudant.prototype.replaceOrCreate %j %j', model, data);
+  var self = this;
+  var idName = self.idName(model);
+  var mo = self.selectModel(model);
+
+  var id = data[idName];
+  if (id) {
+    self.getCurrentRevision(model, id, function(err, rev) {
+      if (err) return cb(err);
+      if (rev) data._rev = rev;
+      self.create(model, data, options, function(err) {
```

Comment:
Now they are almost same, but true `replaceOrCreate` should keep identity to `this._insert()` instead of `this.create()`. Will change it.